float32 angle1
float32 angle2
float32 angle3
